tests/src/python/test_provider_ogr_general.py

```diff
@@ -0,0 +1,980 @@
+        print('-I-> Testing type/values of a \'LINESTRING\' EWKT[{0}] count[{1},{2}]'.format('SRID=3068;POINT(24742.22135058679 20740.69355198127)',vl_linestrings.featureCount(),len(features_vl_linestrings)))
+        test_geom_linestring = [f_iter.geometry() for f_iter in features_vl_linestrings][0].geometry()
+        self.assertEqual(test_geom_linestring.wkbType(), QgsWkbTypes.LineString)
+        test_geom_point_0 = test_geom_linestring.pointN( 0)
+        self.assertEquals((test_geom_point_0.x(), test_geom_point_0.y()), (24742.2213505868,20740.6935519813))
+        test_geom_name = [f_iter['name'] for f_iter in features_vl_linestrings][0]
+        print('-I-> test_geom_linestring[{0}]: name[{0}] first point WKT[{0}]'.format(test_geom_linestring.wktTypeStr(),test_geom_name,test_geom_point_0.asWkt()))
+        del test_geom_point_0
+        del test_geom_linestring
+        del vl_linestrings
+        del features_vl_linestrings
+        # Testing Polygon
+        vl_polygons = QgsVectorLayer(uri_name_polygon, test_name, u'ogr')
+        self.assertTrue(vl_polygons.isValid())
+        features_vl_polygons = [f_iter_polygon for f_iter_polygon in vl_polygons.getFeatures(QgsFeatureRequest().setFilterExpression(u"id_admin=2"))]
+        print('-I-> Testing type/values of a \'POLYGON\' EWKT[{0}] count[{1},{2}]'.format('SRID=3068;POINT(24706.46898365996 20778.82165136906)',vl_polygons.featureCount(),len(features_vl_polygons)))
+        test_geom_polygon = [f_iter_polygon.geometry() for f_iter_polygon in features_vl_polygons][0].geometry()
+        print('-I-> test_geom_polygon[{0}]: name[{0}] first point WKT[{0}]'.format(test_geom_polygon.wktTypeStr(),test_geom_name,test_geom_polygon.asWkt()))
+        self.assertEqual(test_geom_polygon.wkbType(), QgsWkbTypes.Polygon)
+        test_geom_exterior_ring = test_geom_polygon.exteriorRing()
+        test_geom_point_1 = test_geom_exterior_ring.pointN( 0)
+        self.assertEquals((test_geom_point_1.x(), test_geom_point_1.y()), (24706.46898366,20778.8216513691))
+        print('-I-> test_geom_polygon[{0}]: name[{0}] first point WKT[{0}]'.format(test_geom_polygon.wktTypeStr(),test_geom_name,test_geom_point_1.asWkt()))
+        del test_geom_point_1
+        del test_geom_exterior_ring
+        del test_geom_polygon
+        del features_vl_polygons
+        del vl_polygons
+        print('-I-> test_04_{0} [{1}]'.format(test_name,'end of test'))
+
+###############################################################################
+# Sample kmz file included in bug report 15168
+# - https://hub.qgis.org/issues/15168
+# Problem caused by duplicate layername. Retrievel by layername always returned first found layer.
+# - lead to correction: Use "layerid=N" instead of "layername=XYZ" for OGR sublayers
+# which causes problems with gdal 2.*, where layerid logic has changed.
+#  Up to Gdal 2.* each geometry was 1 layer and could be retrieved using an index
+#  Starting wit
```

Comment:
> Problem caused by duplicate layername.
As documents, to check that a layer with a duplicate name can be correctly retrieved.

Resolving https://hub.qgis.org/issues/15168.